devdocs/dependencies.md

```diff
@@ -27,3 +27,12 @@ To propagate the update of dependencies to your Eclipse configuration, run the f
 `./gradlew resetEclipseDeps`
 
 Sometimes, the changes from this command might not show up in Eclipse immediately. "Refreshing" the project or restarting Eclipse should fix that.
+
+# Managing Client-Side Dependencies
+
+Currently, the dependencies for CSS/JS are hosted in the repository and listed down in `src/main/webapp/package.json`.
+
+When adding new libraries, try to find the library in the [npm registry](https://www.npmjs.com) and if it can be found, list it in the aforementioned package file, then download and include the necessary library files (usually in `/dist` or `/build` folder) in the repository.
+Node.js developers can do this with the command `npm install --save package@version`. Remember to exclude any non-exact version syntax (e.g `^` and `~`) in the package file as we require specific versions.
+
+Otherwise, simply host the necessary library files in the repository without updating the package file.
```

Comment:
Also describe the benefits of each option. When reading this part, it's not clear why the library should be added to the package.json.
Do we need the 2nd option at all?